xdsl/dialects/onnx.py

```diff
@@ -419,11 +423,159 @@ def verify_(self) -> None:
             )
 
 
+@irdl_op_definition
+class Conv(IRDLOperation):
+    """
+    The convolution operator consumes an input tensor and a filter, and computes the output.
+
+    Attributes:
+
+    - auto_pad  string (default is NOTSET): auto_pad must be either NOTSET, SAME_UPPER, SAME_LOWER or
+    VALID. Where default value is NOTSET, which means explicit padding is used. SAME_UPPER or SAME_LOWER mean pad the
+    input so that output_shape[i] = ceil(input_shape[i] / strides[i]) for each axis i. The padding is split between
+    the two sides equally or almost equally (depending on whether it is even or odd). In case the padding is an odd
+    number, the extra padding is added at the end for SAME_UPPER and at the beginning for SAME_LOWER.
+
+    -dilations list of ints: dilation value along each spatial axis of the filter. If not present, the dilation
+    defaults is 1 along each spatial axis.
+
+    -group int (default is '1'): number of groups input channels and output channels are divided into.
+
+    -kernel_shape list of ints: The shape of the convolution kernel. If not present, should be inferred from input W.
+
+    -pads list of ints: Padding for the beginning and ending along each spatial axis, it can take any value greater
+    than or equal to 0. The value represent the number of pixels added to the beginning and end part of the
+    corresponding axis. `pads` format should be as follow [x1_begin, x2_begin...x1_end, x2_end,...], where xi_begin
+    the number of pixels added at the beginning of axis `i` and xi_end, the number of pixels added at the end of axis
+    `i`. This attribute cannot be used simultaneously with auto_pad attribute. If not present, the padding defaults
+    to 0 along start and end of each spatial axis.
+
+    -strides list of ints: Stride along each spatial axis. If not present, the stride defaults is 1 along each spatial axis.
+
+    """
+
+    name = "onnx.Conv"
+    T = Annotated[AnyFloat | IntegerType, ConstraintVar("T")]
+    data = operand_def(TensorType[T])
+    weight = operand_def(TensorType[T])
+    bias = operand_def(TensorType[T] | NoneType)
+    res = result_def(TensorType[T])
+
+    auto_pad = attr_def(StringAttr, attr_name="auto_pad")
+    dilations = attr_def(ArrayAttr[AnyIntegerAttr], attr_name="dilations")
+    group = attr_def(AnyIntegerAttr)
+    kernel_shape = attr_def(ArrayAttr[AnyIntegerAttr], attr_name="kernel_shape")
```

Comment:
No need for custom names if they're the same, I'd recommend using them only if there's a mismatch in case style from Python snake_case and cpp camelCase or if there's a name mismatch like `name` or `verify`